beacon_node/eth2-libp2p/src/rpc/handler.rs

```diff
@@ -322,8 +322,10 @@ where
     ) {
         let (req, substream) = substream;
         // drop the stream
-        if let r @ RPCRequest::Goodbye(_) = req {
-            self.events_out.push(RPCEvent::Request(None, r));
+        if let RPCRequest::Goodbye(_) = req {
+            self.events_out
+                .push(RPCReceived::Request(self.current_inbound_substream_id, req));
+            self.current_inbound_substream_id += 1;
```

Comment:
This is interesting. 
We should propagate the goodbye message here as we do in `events_out`. 
Also, we should start the disconnect process from this peer. In this case, we don't want to send back a goodbye. We just want to disconnect.

Reply:
will remove this on a future PR https://github.com/sigp/lighthouse/issues/1240